lib/workers/repository/process/extract-update.ts

```diff
@@ -47,7 +50,36 @@ export async function extract(
   config: RenovateConfig
 ): Promise<Record<string, PackageFile[]>> {
   logger.debug('extract()');
-  const packageFiles = await extractAllDependencies(config);
+  const { baseBranch, baseBranchSha } = config;
+  let packageFiles;
+  const cache = getCache();
+  const cachedExtract = cache?.scan?.[baseBranch];
+  const configHash = hash(config);
+  // istanbul ignore if
+  if (
+    cachedExtract?.sha === baseBranchSha &&
+    cachedExtract?.configHash === configHash
+  ) {
+    logger.debug({ baseBranch, baseBranchSha }, 'Found cached extract');
+    packageFiles = cachedExtract.packageFiles;
+  } else {
+    packageFiles = await extractAllDependencies(config);
+    cache.scan = cache.scan || {};
```

Comment:
```suggestion
    cache.scan = cache.scan || Object.create({});
```

Make sure we don't inherit any properties from `Object` otherwise we can have name collisions.